### DISCUSSION TITLE
My existing files in my repository are gone.

### DISCUSSION THREAD

**Discussion by AKSHAY12332** [community/community#123143] [Repositories]
### Select Topic Area

General

### Body

I force pushed my project folder from local repository to remote repository so, because of that my existing files on the repository were gone. Is there any way to recover those files?

**Comment by CopperEagle** [comment]
Hey @AKSHAY12332 
A repository is not just a set of files, it also has a history. 

### GUI 
You can see the history in a visual way in `(repo) > insights > Network` and you can see the uploads and changes as a graph. You can click a vertex to see this commit. You see a view like the screenshot below. There is a button in the upper right corner, "browse files": 
![browse](https://github.com/community/community/assets/137458753/fd584360-e4d4-482a-b218-126fd446d31b)

Click it and you travel back in time, to the old state of the repository at that commit. Then you can check these files and download them.

### Terminal
If you want to fully *revert* the repository, you can use the terminal for this. [Here](https://stackoverflow.com/questions/4114095/how-do-i-revert-a-git-repository-to-a-previous-commit) is how.

Hope this helps you. If you have any other questions, let the community know. If this answers your question, please consider marking it as the answer, thanks!

**Comment by iTechTop** [comment]
🤔 Generally speaking, once you've force pushed and reset the branch remotely, it's difficult to recover the previously committed files unless you had enabled versioning or backups elsewhere.

There are a few places you can check to potentially recover the files:

1. Local clone of the remote repository: If you had a local clone of the remote repository, you can dig through its history to find the last working commit. You can compare the differences between the forced commit and the older commits to recover the missing files.

2. Branch Protection Rules: Review the branch protection settings in the remote repository. If you had enabled the requirement of a pull request to merge changes, there might be chances that the files existed in the target branch.

3. Personal Access Tokens or Application Passwords: If you utilized any tokens or passwords to connect to the remote repository, there might be traces of the older version of files in the token's connected repositories or credentials manager.

4. Third-Party Integrations: Investigate any third-party integrations that you had set up with the remote repository (e.g., GitHub Actions, CircleCI, Jenkins, or similar systems). Those builds might have kept cached versions of files, allowing you to download and recover the missing files.

Unfortunately, if none of these methods yield positive results, the files are likely lost forever. Going forward, remember to exercise caution when force pushing and consider using backup mechanisms (e.g., archiving zipballs, snapshots, periodic backups, etc.) to safeguard yourself against accidental deletions.
